File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.swift
class Solution {
    

    func lengthOfLongestSubstring(_ s: String) -> Int {
        
        if s.count == 0 || s.count == 1 {
            return s.count
        }

        var maxLength = 0
        var substring: [Character] = []
        let charArray = Array(s)
        
        for i in 0 ..< charArray.count {
            if let index = substring.firstIndex(of: charArray[i]) {
                substring.removeFirst(index + 1)
            }
            
            substring.append(charArray[i])
            
            print(substring)
            maxLength = max(maxLength, substring.count)
        }
        
        
        return maxLength
        
        
        
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        
        
        int max = 0;
        
        string substring = "";
        
        int left = 0;
        int right = 0;
        
        while(right < s.Length) {
            
            if (substring.Contains(s[right])) {

                left = substring.IndexOf(s[right]) + 1;
                substring = substring.Substring(left);  
                continue;
            }
            
            substring += s[right];
            
            max = Math.Max(max, substring.Length);
            right++;
            
        }
        
        
        
        
        return max;
            
    }
}